### PR TITLE
[WIP] Add a `get_token_embeddings_size` to `PreTrainedModel`

### LINKED ISSUE #20377: PreTrainedModel: provide a more intuitive way of getting the current size of embeddings
### Feature request

Please add an intuitive and obvious method to get the size of the current token embeddings on `PreTrainedModel`.

For example, a method called `get_token_embeddings_size()` that can be used as a complement to `resize_token_embeddings()`.

### Motivation

Current API design around the `resize_token_embeddings()` method requires doing the following to increase the size of the token embeddings by 1:

```python
# add 1 new embedding
current_embeddings = my_xformer.resize_token_embeddings(None)
new_embeddings_size = current_embeddings.num_embeddings + 1
my_xformer.resize_token_embeddings(new_embeddings_size)
```

This is counterintuitive and bleeds implementation details to the call site. It requires me to know
1. that calling a "resize" method with the argument `None` returns the object to be resized (which is not intuitive), and
2. that "size" means the property `num_embeddings` on the returned object (admittedly it's not difficult to guess, but it is still a *guess*, and is in fact an implementation detail that I shouldn't need to know).

It would be better if I could do something like this:

```python
# add 1 new embedding
current_embeddings_size = my_xformer.get_token_embeddings_size()
new_embeddings_size = current_embeddings_size + 1
my_xformer.resize_token_embeddings(new_embeddings_size)
```

This provides an intuitively-named method to determine the current "size", and appropriately hides the implementation detail that "size" means the `num_embeddings` property on the object to be resized.

### Your contribution

Here's a proposed implementation:

```python
    def get_token_embeddings_size(self) -> int:
        return model_embeds.num_embeddings
```

i can make a PR if it would help.

### PR DESCRIPTION
**Pull request by damian0815**
Adds an intuitive and obvious method, `get_token_embeddings_size()`, to get the size of the current token embeddings on `PreTrainedModel`. This can be used to compute the new size when calling `resize_token_embeddings()`.

### Motivation

Current API design around the `resize_token_embeddings()` method requires doing the following to increase the size of the token embeddings by 1:

```python
# add 1 new embedding
current_embeddings = my_xformer.resize_token_embeddings(None)
new_embeddings_size = current_embeddings.num_embeddings + 1
my_xformer.resize_token_embeddings(new_embeddings_size)
```

This is counterintuitive and bleeds implementation details to the call site. It requires me to know
1. that calling a "resize" method with the argument `None` returns the object to be resized (which is not intuitive), and
2. that "size" means the property `num_embeddings` on the returned object (admittedly it's not difficult to guess, but it is still a *guess*, and is in fact an implementation detail that I shouldn't need to know).

# What does this PR do?

This PR enables instead the following:

```python
# add 1 new embedding
current_embeddings_size = my_xformer.get_token_embeddings_size()
new_embeddings_size = current_embeddings_size + 1
my_xformer.resize_token_embeddings(new_embeddings_size)
```

This provides an intuitively-named method to determine the current "size", and appropriately hides the implementation detail that "size" means the `num_embeddings` property on the object to be resized.

Fixes #20377 .

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [*] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [*] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case. - #20377 .
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by damian0815** [comment]
WIP because missing tests, i guess. I don't have a local environment set up atm, this was a quick edit on github.

File: src/transformers/modeling_utils.py [hunk 1]
```diff
@@ -1215,10 +1215,31 @@ def _tie_or_clone_weights(self, output_embeddings, input_embeddings):
         if hasattr(output_embeddings, "out_features") and hasattr(input_embeddings, "num_embeddings"):
             output_embeddings.out_features = input_embeddings.num_embeddings
 
+    def get_token_embeddings_size(self) -> int:
+        """
+        Return the current size of the input token embeddings matrix. 
+        
+        Use in conjunction with resize_token_embeddings to provide a baseline against which size changes can be
```
Comment (sgugger) [inline src/transformers/modeling_utils.py:1222]:
```suggestion
        Use in conjunction with `resize_token_embeddings` to provide a baseline against which size changes can be 
```

**Review by sgugger — COMMENTED** [review]
Thanks for your PR! I have a couple of nits on the docs. Make sure to run `make style` on your branch so that the quality test pass.

**Comment by damian0815** [comment]
i still am interested in finishing this, just gotta find some time.

**Pull request closed without merging**